src/source/geojson_source.js

```diff
@@ -276,24 +250,28 @@ class GeoJSONSource extends Evented implements Source {
         // target {this.type}.loadData rather than literally geojson.loadData,
         // so that other geojson-like source types can easily reuse this
         // implementation
-        this.actor.send(`${this.type}.loadData`, options, (err, result) => {
-            if (this._removed || (result && result.abandoned)) {
+        this._pendingLoad = this.actor.send(`${this.type}.loadData`, options, (err, result) => {
+            this._loaded = true;
+            this._pendingLoad = null;
+
+            if (err) {
+                this.fire(new ErrorEvent(err));
                 return;
```

Comment:
An error here will prevent the `_updateWorkerData()` call when `_coalesce = true`. Does this matter?

Reply:
Nice catch! It's a rare occasion (e.g. running `setData` with a malformed GeoJSON and then following it shortly with a correct one), but fixed it just in case.